lib/rubygems/platform.rb

```diff
@@ -30,6 +30,20 @@ def self.match(platform)
     end
   end
 
+  def self.rank(platform)
```

Comment:
is there any reason this sorts based upon the local platform, rather than being consistent across installs? See, for example, `Bundler::Resolver.sort_platforms` for a different approach

Reply:
> is there any reason this sorts based upon the local platform

Yes, because it's determining what platform is selected for install/update.  Note that it's used in `max_by` selection.

Again, if you go the end of the method and insert a line `nil` at line 45, you'll see some of the new tests failing.